### PR TITLE
Linter: Add warning message [WIP]

### PR DESCRIPTION
**Pull request by mr-karan**
Add warning message in the `run()` method
which will check whether the language is
supported by Bear or not. Prints a list
of bears that user can use for that file
if not supported.

Fixes https://github.com/coala-analyzer/coala/issues/2309

### CODE REVIEW

**Comment by mr-karan** [comment]
https://asciinema.org/a/48487

File: coalib/bearlib/abstractions/Linter.py [hunk 1]
```diff
@@ -467,6 +486,38 @@ def _create_config(cls, filename, file, **kwargs):
                     yield config_file
 
         def run(self, filename, file, **kwargs):
+            printer = ConsolePrinter()
+            log_printer = LogPrinter(NullPrinter())
+            sections, _ = load_configuration(None, log_printer)
+            local_bears, global_bears = collect_all_bears_from_sections(
+                sections, log_printer)
```
Comment (sils1297) [inline coalib/bearlib/abstractions/Linter.py]: this is really bad for performance, plus we already have the bears collected in coala. Maybe this check should rather happen in coala itself?

As pointed out on gitter, the linter is the wrong place, we'll want it for *all* bears.

File: coalib/bearlib/abstractions/Linter.py [hunk 2]
```diff
@@ -467,6 +486,38 @@ def _create_config(cls, filename, file, **kwargs):
...
+            ext = os.path.splitext(filename)[1]
+            lang_map = {"all": "All"}
+            lang_map[exts[ext].lower()] = exts[ext]
+            relevant_bears = {"All": []}
+            relevant_bears[exts[ext]] = []
+            for bear in bears:
+                if isinstance(bear.LANGUAGES, tuple):
+                    bear_languages = [lang.lower() for lang in bear.LANGUAGES]
+                else:
+                    bear_languages = [bear.LANGUAGES.lower()]
```
Comment (sils1297) [inline coalib/bearlib/abstractions/Linter.py]: just use the `bear.supported_languages` property which is guaranteed to be a tuple

**Comment by Makman2** [comment]
do we still want it?

**Comment by sils1297** [comment]
I'm against that

**Comment by Makman2** [comment]
me too
@mr-karan ?

**Comment by mr-karan** [comment]
@hypothesist has updated his Extensions.py which now supports multiple languages. https://github.com/coala-analyzer/coala-quickstart/blob/hypothesist/wip/coala_quickstart/Extensions.py

Otherwise the issue status has been changed to `wontfix/not happening` so we can close the PR

**Comment by sils1297** [comment]
@mr-karan if you want to discuss this further (if it's part of your project) you should set up a meeting with your mentor and me and @Makman2 I guess on mumble to wrap this up?

**Pull request closed without merging**